test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +87,320 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+func TestSnapshottable(l *snapshottableLocal, sct SnapshotClassTest) {
+	var err error
+
+	ginkgo.By("creating a SnapshotClass")
+	l.vsc.Object["deletionPolicy"] = sct.DeletionPolicy
+	l.vsc, err = l.dc.Resource(SnapshotClassGVR).Create(context.TODO(), l.vsc, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	l.cleanupSteps = append(l.cleanupSteps, func() {
+		framework.Logf("deleting SnapshotClass %s", l.vsc.GetName())
+		l.dc.Resource(SnapshotClassGVR).Delete(context.TODO(), l.vsc.GetName(), metav1.DeleteOptions{})
 	})
+	l.vsc, err = l.dc.Resource(SnapshotClassGVR).Namespace(l.vsc.GetNamespace()).Get(context.TODO(), l.vsc.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("creating a snapshot")
+	l.vs = getSnapshot(l.pvc.Name, l.pvc.Namespace, l.vsc.GetName())
+
+	l.vs, err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Create(context.TODO(), l.vs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	l.cleanupSteps = append(l.cleanupSteps, func() {
+		framework.Logf("deleting snapshot %q/%q", l.vs.GetNamespace(), l.vs.GetName())
+		// typically this snapshot has already been deleted
+		err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %q. Error: %v", l.pvc.Name, err)
+		}
+	})
+	err = WaitForSnapshotReady(l.dc, l.vs.GetNamespace(), l.vs.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+	framework.ExpectNoError(err)
+
+	ginkgo.By("checking the snapshot")
+	// Get new copy of the snapshot
+	l.vs, err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Get(context.TODO(), l.vs.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	// Get the bound snapshotContent
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+	l.vscontent, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	snapshotContentSpec := l.vscontent.Object["spec"].(map[string]interface{})
+	volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+
+	// Check SnapshotContent properties
+	ginkgo.By("checking the SnapshotContent")
+	framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], l.vsc.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["name"], l.vs.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["namespace"], l.vs.GetNamespace())
+}
+
+// TestSnapshotDeleted tests the results of deleting a VolumeSnapshot
+// depending on the deletion policy currently set.
+func TestSnapshotDeleted(l *snapshottableLocal, sct SnapshotClassTest) {
+	var err error
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+
```

Comment:
Lets put a print here to indicate the status and bound content of the snapshot, for better debugging?